README.md

```diff
@@ -8,31 +8,104 @@
 The `xrpl-go` library provides a Go implementation for interacting with the XRP Ledger. From serialization to signing transactions, the library allows users to work with the most
 complex elements of the XRP Ledger. A full library of models for all transactions and core server rippled API objects are provided.
 
-## Important Information
-This library is still in development and not yet intended for use.
+## Disclaimer
+This library is still in development and not yet intended for production use.
 
-## Installation
-The latest version can be installed using - **TODO**
+## Requirements
 
-Requiring Go version `1.18.0` and later.
+Requiring Go version `1.21.0` and later.
 [Download latest Go version](https://go.dev/dl/)
 
-## Features
-* Binary Codec
-* Address Codec
-* Keypairs Generation - **to finish**
-* Go Models - **to finish**
-* Json Rpc Client - **to finish**
-* Webhook Client - **todo**
+## Packages
+
+| Name | Description |
+|---------|-------------|
+| addresscodec | Provides functions for encoding and decoding XRP Ledger addresses |
+| binarycodec | Implements binary serialization and deserialization of XRP Ledger objects |
+| keypairs | Handles generation and management of cryptographic key pairs for XRP Ledger accounts |
+| xrpl | Core package containing the main functionality for interacting with the XRP Ledger |
+| examples | Contains example code demonstrating usage of the xrpl-go library |
 
 ## Usage
-* Keys and Wallets
-* Serialize and sign transactions
-* Encode addresses into and from classic address (***no support for X-addresses***)
-* How to use with JsonRpc Client
-* How to use Webhooks
 
-## Raising Issues
+Here's a list of examples of how to use the library.
+
+### Create a wallet
+This example shows how to create a new wallet with a random seed or from a given seed.
+
+```go
+wallet, err := xrpl.NewWallet(addresscodec.ED25519)
+walletFromSeed, _ := xrpl.NewWalletFromSeed("sEdSMVV4dJ1JbdBxmakRR4Puu3XVZz2", "")
+walletFromSecret, _ := xrpl.NewWalletFromSecret("sEdSMVV4dJ1JbdBxmakRR4Puu3XVZz2")
+```
+
+### Fund a wallet
```

Comment:
Would it be possible to have two sections for the clients ? One for the JSON RPC and another one for the Websocket.